test/testcases/test_sdk_api/test_dataset_mangement/test_list_datasets.py

```
@@ -21,6 +21,55 @@
+
+    @pytest.mark.p2
+    @pytest.mark.parametrize(
+        "params",
+        [{"id": "dataset-id", "ids": ["dataset-id"]}, {"id": "dataset-id", "ids": [""]}],
+    )
+    def test_rejects_conflicting_id_and_ids_before_get(self, monkeypatch, params):
+        client = RAGFlow("token", HOST_ADDRESS)
+        get_calls = []
+        monkeypatch.setattr(client, "get", lambda *_args, **_kwargs: get_calls.append((_args, _kwargs)))
+
+        with pytest.raises(ValueError, match="^Cannot use both 'id' and 'ids' parameters at the same time\\.$"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)